.github/cloud-samples-tools/pkg/utils/config.go

```diff
@@ -0,0 +1,137 @@
+/*
+ Copyright 2024 Google LLC
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      https://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package utils
+
+import (
+	"encoding/json"
+	"errors"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+)
+
+type Config struct {
+	// Filename to look for the root of a package.
+	PackageFile []string `json:"package-file"`
+
+	// Pattern to match filenames or directories.
+	Match []string `json:"match"`
+
+	// Pattern to ignore filenames or directories.
+	Ignore []string `json:"ignore"`
+
+	// Packages to always exclude.
+	ExcludePackages []string `json:"exclude-packages"`
+}
+
+var multiLineCommentsRegex = regexp.MustCompile(`(?s)\s*/\*.*?\*/`)
+var singleLineCommentsRegex = regexp.MustCompile(`\s*//.*\s*`)
+
+// SaveConfig saves the config to the given file.
+func SaveConfig(config Config, file *os.File) error {
```

Comment:
nit: make this a method of `Config` struct, perhaps with a pointer receiver (shouldn't matter, but semantically you're saving the thing, not a copy of it).

Reply:
Good point, I also converted other functions receiving a config to be methods on Config.